File: amd/flang-device-schedule/map.f90
! schedule(static,C) prescribes which thread runs which iteration. Record the
! mapping and print it; if it does not change with C, the clause is ignored.
program p
  use omp_lib
  implicit none
  integer, parameter :: n = 32
  integer :: tid(n), i
  tid = -1
  !$omp target teams distribute parallel do num_teams(1) thread_limit(8) SCHEDCLAUSE map(tofrom:tid)
  do i = 1, n
     tid(i) = omp_get_thread_num()
  end do
  write(*,'(A,32I3)') "  tid: ", tid
end program

File: amd/flang-device-schedule/map.c
#include <stdio.h>
#include <omp.h>
int main(void) {
  enum { n = 32 };
  static int tid[n];
  for (int i = 0; i < n; ++i) tid[i] = -1;
  #pragma omp target teams distribute parallel for num_teams(1) thread_limit(8) SCHEDCLAUSE map(tofrom: tid)
  for (int i = 0; i < n; ++i) tid[i] = omp_get_thread_num();
  for (int i = 0; i < n; ++i) printf("%3d", tid[i]);
  printf("\n");
  return 0;
}
